python_anvil/api_resources/mutations/create_etch_packet.py

```diff
@@ -206,7 +202,41 @@ def get_file_payloads(self):
                 )
         return self.file_payloads
 
-    def create_payload(self) -> CreateEtchPacketPayload:
+    def get_files_for_payload(self) -> Tuple[List[AttachableEtchFile], List[Any]]:
+        """Scan through and gather files for use in multipart requests.
+
+        During the process, files will be replaced with a dummy `Path`
+        instance as `Callable` instances and likely their returns of some
+        sort of IO reader is not serializable and not easily dealt with in
+        pydantic (the underlying model system for the types used in this
+        library). Closer to the actual request, when the multipart payload is
+        being constructed, the dummy instances will be replaced by their actual
+        file instances.
+
+        :return: Tuple containing a list of uploadable file types, and a list
+            of actual file references
+        :rtype: Tuple[List, List]
+        """
+        contains_uploads = any([isinstance(f, DocumentUpload) for f in self.files])
+        if not contains_uploads:
+            return self.files, []
+
+        files = []
+        for idx, f in enumerate(self.files):
+            if not isinstance(f, DocumentUpload):
+                continue
+
+            if getattr(f.Config, "contains_uploads", False):
+                attr_name = getattr(f.Config, "contains_uploads")
+                upload = getattr(f, attr_name)
+
+                if not isinstance(upload, Base64Upload):
+                    files.append(upload)
+                    setattr(f, attr_name, Path())
+
+        return self.files, files
```

Comment:
Admittedly, this is a little funky, but it's the easiest way I can think of at the moment. Because of the multipart spec, we need to check for files, but the file attribute also has to be nulled/`None`-ed out when we serialize this down to JSON.